creme/utils/shift.py

```diff
@@ -0,0 +1,100 @@
+import collections
+import typing
+
+__all__ = ['Shift']
+
+
+class DefaultStat:
+    """Default stat when using Shift class"""
+
+    def __init__(self):
+        self.x = None
+
+    def update(self, x):
+        self.x = x
+        return self
+
+    def get(self):
+        return self.x
+
+    @property
+    def name(self):
+        return ''
+
+
+class Shift:
+    """Shift the observations with the selected period.
+
+    Shift class allows to store and recall at regular intervals values observed in a stream.
+    The method ```get()`` returns the ```n-th - period``` observed entry.
+
+    Parameters:
+        period (int): Number of periods to shift. Must be positive.
+        stats (stats.Univariate, None): The statistic to compute. When set to None, shift the input
+            series of values without updating any statistics.
+        missing: Handle missing values when there is'nt any reference for a given period.
+        fitting_before_transform (bool): Shift logic. Must be set to True when the user updates the
+            buffer before retrieving the observation with the offset selected.
+
+    Example:
+
+        ::
```

Comment:
We don't need to do this anymore. A single tab suffices. Check out the other docstrings :)